packages/massa-web3/src/experimental/smartContract.ts

```diff
@@ -7,38 +7,48 @@ import {
   CallOperation,
 } from './basicElements'
 import * as StorageCost from './basicElements/storage'
-import { BlockchainClient } from './client'
+import { BlockchainClient, ReadOnlyCallResult } from './client'
 import { Account } from './account'
 import { ErrorInsufficientBalance, ErrorMaxGas } from './errors'
 import { deployer } from './generated/deployer-bytecode'
 import { populateDatastore } from '../../deployer_generation/dataStore'
 import { ONE, ZERO } from './utils'
 import { execute } from './basicElements/bytecode'
+import { Mas } from './basicElements/mas'
 
-// TODO: Move to constants file
 export const MAX_GAS_CALL = 4294167295n
 export const MIN_GAS_CALL = 2100000n
+export const MAX_GAS_DEPLOYMENT = 3980167295n
+
 const DEFAULT_PERIODS_TO_LIVE = 9
 
 type CommonOptions = {
-  fee?: bigint
+  fee?: Mas
   maxGas?: bigint
-  coins?: bigint
+  coins?: Mas
   periodToLive?: number
 }
 
 type DeployOptions = {
-  fee?: bigint
+  fee?: Mas
   maxGas?: bigint
-  maxCoins?: bigint
+  maxCoins?: Mas
   periodToLive?: number
   waitFinalExecution?: boolean
 }
 
+type ReadOnlyCallOptions = {
+  parameter?: Uint8Array
```

Comment:
It should be an argument of the `read` function. Even if it won't always be necessary, you can set a default value if you want, since in most cases we call a function with parameters.